### ISSUE TITLE
MitmProxy crashes

### ISSUE THREAD

**Issue by kuza55** [mitmproxy/mitmproxy#1339]
##### Steps to reproduce the problem:

I'm not sure what is causing this error, I was running mitmdump to mitm chrome like so:

mitmdump --stream 10k -q -s "myscript.py"

I don't get this error without both --stream and -q

##### What went wrong?
I got this stack trace:

```
127.0.0.1:50357: Traceback (most recent call last):
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\proxy\server.py", line 121, in handle
    root_layer()
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\proxy\modes\http_proxy.py", line 11, in __call__
    layer()
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\protocol\tls.py", line 358, in __call__
    layer()
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\protocol\http1.py", line 67, in __call__
    layer()
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\protocol\http.py", line 171, in __call__
    self.handle_regular_mode_connect(request)
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\protocol\http.py", line 263, in handle_regular_mode_connect
    layer()
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\protocol\tls.py", line 358, in __call__
    layer()
  File "c:\users\alex\anaconda2\lib\site-packages\mitmproxy\protocol\http2.py", line 234, in __call__
    events = source_conn.h2.receive_data(raw_frame)
  File "c:\users\alex\anaconda2\lib\site-packages\h2\connection.py", line 1143, in receive_data
    events.extend(self._receive_frame(frame))
  File "c:\users\alex\anaconda2\lib\site-packages\h2\connection.py", line 1166, in _receive_frame
    frames, events = self._frame_dispatch_table[frame.__class__](frame)
  File "c:\users\alex\anaconda2\lib\site-packages\h2\connection.py", line 1234, in _receive_headers_frame
    headers = validate_headers(headers)
  File "c:\users\alex\anaconda2\lib\site-packages\h2\utilities.py", line 167, in validate_headers
    return list(headers)
  File "c:\users\alex\anaconda2\lib\site-packages\h2\utilities.py", line 221, in _reject_pseudo_header_fields
    for header in headers:
  File "c:\users\alex\anaconda2\lib\site-packages\h2\utilities.py", line 206, in _reject_connection_header
    "Connection-specific header field present: %s." % header[0]
ProtocolError: Connection-specific header field present: connection.
mitmproxy has crashed!
```

**Comment by mhils** [comment]
Which mitmproxy version is that?

**Comment by kuza55** [comment]
mitmdump 0.17

**Comment by Kriechi** [comment]
This should already be fixed with 05de3b759fe75e2f6839d8e855c304239f97eec1.
@kuza55 could you try installing from git-master and check again? Thanks.
